opencog/truthvalue/AttentionValue.h

```diff
@@ -35,8 +35,10 @@ namespace opencog
  */
 
 //! stores attention in three components: short-term, long-term and very long-term
+//! This class is currently read-only, and assumed to be thread-safe. Beware of
+//! possible race conditions if it is changed
 class AttentionValue;
-typedef std::shared_ptr<AttentionValue> AttentionValuePtr;
+typedef std::shared_ptr<const AttentionValue> AttentionValuePtr;
```

Comment:
Ah, well, if the goal is to have atomic thread-safety, then this should be achieved in the atoms, and we should do the same for truth values, as well. I am pretty use that just marking the pointer itself as const probably breaks the scheme, python and presumably the haskell bindings too.  Let me take a look.